backend/app/api/api_v1/endpoints/job.py

```diff
@@ -0,0 +1,111 @@
+from typing import Any, List
+
+from fastapi import APIRouter, Body, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+from app import crud, schemas
+from app.api import deps
+from app.core.config import settings
+
+router = APIRouter()
+
+@router.get("/all", response_model=List[schemas.Job])
+def get_all_job(
+    db: Session = Depends(deps.get_db),
+    skip: int = 0,
+) -> Any:
+    """
+    Retrieve all jobs.
+    """
+    jobs = crud.job.get_multi(db, skip=skip)
+    if not jobs:
+        raise HTTPException(
+            status_code=404,
+            detail="Jobs not found",
+        )
+    return jobs
+
+@router.get("/{job_id}", response_model=schemas.Job)
+def get_job_by_id(
+    job_id: int,
+    db: Session = Depends(deps.get_db),
+) -> Any:
+    """
+    Get a specific job by id.
+    """
+    job = crud.job.get(db, job_id=job_id)
+    if not job:
+        raise HTTPException(
+            status_code=404,
+            detail="Job not found",
+        )
+    return job
+
+@router.post("", response_model=schemas.Job)
+def create_job(
+    *,
+    db: Session = Depends(deps.get_db),
+    job_name: str = Body(...),
+    job_desc: str = Body(...),
```

Comment:
In the alembic SQL, `job_desc` does not have a `NOT NULL` constraint. In this case, using `Body(...)` would enforce the field to be a compulsory field in the body. If it's not compulsory, it should be `Body(None)` instead

Reply:
changed!